### PR TITLE
[WIP] - 40 add view previous auction functionality on the home page

### LINKED ISSUE #40: add view previous auction functionality on the home page
without getting into the full gov interface workflow, another low hanging fruit for the home page is prob the addition of back/forward arrows to see previous auction results. see screenshot below for example on nouns.wtf

Screen Shot 2022-12-04 at 8 14 02 PM

for our implementation, I believe we could just throw the arrows below the "Public Assembly #x" ttitle and then add another row of blank space below. when viewing tokens that have alrady been distributed, I think we need to add in the following info:

1. who owns that token (which isnt necessarily the same as who won the auction)
2. how much did that token sell for. if it was a founders allocated token, that should be made apparent, rather than just having a weird 0eth / N/A price view

Screen Shot 2022-12-04 at 8 15 24 PM

### PR DESCRIPTION
**Pull request by dblodorn**
Adding increment / decrement - token history, winning bid, etc.

Screen Shot 2022-12-12 at 1 54 12 PM

### CODE REVIEW

File: @dao-auction/hooks/useNounsProtocol.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import * as React from 'react'
```
Comment (dblodorn) [inline @dao-auction/hooks/useNounsProtocol.ts:1]: @0xTranqui @Javier-Szyfer - so typechain is pretty cool for interacting with contracts - removes a lot of guesswork, and you can sculpt your contract calls in pretty custom ways as opposed to wagmi.

File: @dao-auction/components/TokenWinningBid.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  React.useEffect(() => {
+    async function getBids() {
+      try {
+        if(tokenData?.mintInfo?.mintContext?.blockNumber) {
+          /** 
+           * https://docs.ethers.io/v5/api/contract/contract/#Contract-queryFilter
+           * Used to query the Auction events exposed below:
+           * https://github.com/ourzora/nouns-protocol/blob/main/src/auction/IAuction.sol#L16-L22
+           */
+          const bids = await BuilderAuction?.queryFilter(
```
Comment (dblodorn) [inline @dao-auction/components/TokenWinningBid.tsx:40]: @0xTranqui / @salieflewis this ethers read call is soooo cool...

Comment (salieflewis) [reply @dao-auction/components/TokenWinningBid.tsx:40]: Did you write `useNounsProtocol`, `useActiveAuction`?

Comment (dblodorn) [reply @dao-auction/components/TokenWinningBid.tsx:40]: Yes.

**Comment by dblodorn** [comment]
@0xTranqui @salieflewis - cool if i just merge this.
There's much more to do to smooth it all out... like:

- not fully in line with the design in figma
- I think i need to move away from the zora indexer for auction events, it was out of sync from what i understand during the end, this may have to do with auction extensions... or the indexer trailing events a little.

to fix stuff i need to update the structure of what i've written a bit to utilize a provider / provider-hook pattern so the UI is more composable (ie. forward back arrows), and then swap out the zora api calls for direct contract reads... But i need to set up a contract with quick auction turnover to test with... until we actually start writing real tests for frontend stuff / :

**Comment by salieflewis** [comment]
I came across some design inspiration from this [design studio](https://partnerandpartners.com/) I really admire. I feel like this is where we're headed in terms of aesthetic, and I think we just need to lean into it if we're going to go this way. Totally cool with the design decisions you made in this PR.

And I hear you about updating the hooks. We could also theoretically make a subgraph for our contract to be right on top of our events, but obviously that won't scale for other DAOs.

**Pull request merged**